llvm/utils/TableGen/CodeEmitterGen.cpp

```diff
@@ -290,11 +290,55 @@ CodeEmitterGen::getInstructionCases(Record *R, CodeGenTarget &Target) {
     if (auto *DI = dyn_cast_or_null<DefInit>(RV->getValue())) {
       const CodeGenHwModes &HWM = Target.getHwModes();
       EncodingInfoByHwMode EBM(DI->getDef(), HWM);
+      unsigned EncodingHwModesInBits = DefaultMode;
+      for (auto &[ModeId, Encoding] : EBM) {
+        // DefaultMode is 0, skip it.
+        if (ModeId != DefaultMode)
+          EncodingHwModesInBits |= (1 << (ModeId - 1));
+      }
+
+      // Get HwModes for this Instr by bitwise AND operations,
+      // and find the table to which this instr and hwmode belong.
```

Comment:
Nit: Be consistent in the prose-- we use `Instr` here and then `instr` there, and `HwModes` here vs `hwmode` there. And just say "mask" rather than the long-winded "by bitwise AND oeprations".

 I'd probably just write "Get the applicable hardware modes for this instruction by masking off the relevant bits."